File: modules/core/src/test/java/kr/entree/enderwand/CommandTest.java
package kr.entree.enderwand;

import kr.entree.enderwand.collection.Reader;
import kr.entree.enderwand.command.Argument;
import kr.entree.enderwand.command.Command;
import kr.entree.enderwand.command.CommandBuilder;
import kr.entree.enderwand.command.CommandContext;
import kr.entree.enderwand.command.sender.ConsoleSender;
import kr.entree.enderwand.command.sender.Sender;
import org.junit.jupiter.api.Test;

import java.util.concurrent.atomic.AtomicBoolean;
import java.util.concurrent.atomic.AtomicInteger;

import static org.junit.jupiter.api.Assertions.assertTrue;

/**
 * Created by JunHyung Lim on 2020-02-04
 */
public class CommandTest {
    @Test
    public void simple() {
        AtomicBoolean invoked = new AtomicBoolean();
        Command<Sender> command = new CommandBuilder<>()
                .executes(ctx -> invoked.set(true))
                .build();
        command.execute(new CommandContext<>(ConsoleSender.INSTANCE, Reader.of()));
        assertTrue(invoked.get());
    }

    Argument<Number> number() {
        return new Argument<>();
    }

    @Test
    public void composition() {
        String input = "1 plus 2";
        AtomicInteger result = new AtomicInteger();
        // TODO
    }
}

File: modules/core/src/test/kotlin/kr/entree/enderwand/CommandTest.kt
package kr.entree.enderwand

import kr.entree.enderwand.collection.toReader
import kr.entree.enderwand.command.command
import kr.entree.enderwand.command.executor.execute
import kr.entree.enderwand.command.sender.ConsoleSender
import kr.entree.enderwand.command.sender.Sender
import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertTrue
import org.junit.jupiter.api.Test

/**
 * Created by JunHyung Lim on 2020-01-09
 */
class CommandTest {
    @Test
    fun simple() {
        val msg = "Done!"
        var result = 0
        var msgReceived = false
        command<Sender> {
            executor {
                result = args[0].toInt() + args[1].toInt()
                sender.sendMessage(msg)
            }
        }.execute(createSender { msgReceived = it == msg }, listOf("2", "3").toReader())
        assertEquals(5, result)
        assertTrue(msgReceived)
    }
}